File: implementations/node/sdk/src/contracts/pipelines/invocation-definition.ts
import type { AiSdkInvocationKind } from "./invocation-kind.js";

export interface AiSdkInvocationDefinition {
  readonly kind: AiSdkInvocationKind;
  readonly implementationRef?: string;
  readonly connectionRef?: string;
  readonly tool?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/pipelines/invocation_definition.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .invocation_kind import AiSdkInvocationKind


@dataclass(frozen=True)
class AiSdkInvocationDefinition(AiSdkWireModel):
    kind: AiSdkInvocationKind
    implementation_ref: str | None = None
    connection_ref: str | None = None
    tool: str | None = None
